### ISSUE TITLE
Get program name from command line when invoked as package

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#600]
*Originally reported by* **Ben Finney (Bitbucket: [bignose](https://bitbucket.org/bignose), GitHub: [bignose](https://github.com/bignose))**

----------------------------------------

The program name, for use in the help message, is derived from the command-line arguments — except when the command-line shows the program is the main module of a package.

```python
        self.program_name = os.path.basename(sys.argv[0])
        if self.program_name == '__main__.py':
            self.program_name = 'coverage'
```

The name ‘coverage’ is hard-coded, regardless of what the package actually is.

Instead, when the command-line first argument shows the program was invoked as a package, the package name should be used for the program name.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/600
- This issue had attachments: 0001-Get-the-command-name-from-the-file-path-of-the-first.patch.  See the [original issue](https://bitbucket.org/ned/coveragepy/issue/600) for details.

**Comment by nedbat** [comment]
*Original comment by* **Ben Finney (Bitbucket: [bignose](https://bitbucket.org/bignose), GitHub: [bignose](https://github.com/bignose))**

----------------------------------------

This patch implements a proposed improvement for this bug.

I am not able to figure out how to get the tests running, so I have written test cases that I *hope* demonstrate the problem and test the implementation.

**Comment by nedbat** [comment]
Applied in [975306de6b68 (bb)](https://bitbucket.org/ned/coveragepy/commits/975306de6b68) and fixed in [822e09154bb7 (bb)](https://bitbucket.org/ned/coveragepy/commits/822e09154bb7).  Thanks :)

**Comment by nedbat** [comment]
This has shipped in 4.4.2
